### DISCUSSION TITLE
Firmware Update Issue

### DISCUSSION THREAD

**Discussion by Paramjit26** [bitcraze/discussions#797] [General]
Hello!

I am new to crazyflie platform. After assembling the crazyflie tried to attach the Decks (Multi-Ranger and flow) to it. But when I tried to Upgrade the firmware. It was not able to detect the decks and upgrade stopped in between! 

What Should I do in this situation?

Your Sincerely,
Paramjit Singh Kainth

**Comment by krichardsson** [comment]
Hi,
As part of the firmware upgrade process, the firmware in decks is also updated. Most decks don't have any firmware though and are simply skipped. This is most likely what is happening in your case, the multi ranger and flow decks do not have any firmware so I think all is fine. 

The expected behavior is that the client first updates the STM and NRF firmware in the Crazyflie. After a reboot of the Crazyflie it checks for decks that should be updated (none in your case) and then reboots the Crazyflie again.

**Reply by noahzn** [reply]
Hi, after the first reboot, if the decks do not need to be updated, will the crazyflie reboot again?

**Comment by Paramjit26** [comment]
Hi,

I tried Everything, When I turned on my crazyflie only M2 LED is solid light.  I don't know how to reboot the crazyflie, Is there any process how to reboot the crazyflie.

**Comment by Paramjit26** [comment]
![crazyflie boot Loading](https://github.com/bitcraze/discussions/assets/137099045/03ab0acf-c225-4ec6-ae17-840b833652a6)

**Reply by Paramjit26** [reply]
only Blue M2 lead is solid! I don't know what is the problem

**Reply by Paramjit26** [reply]
https://github.com/bitcraze/discussions/assets/137099045/97590267-26f7-429c-86ae-2bf6d13aefe3

**Reply by ataffanel** [reply]
It looks like you indeed need to re-flash the STM32 and so the cold-boot procedure is the one you should run. For some reason though, the flashing process does not start in your case. On the Crazyflie side there is nothing wrong: the bootloader is well started on both CPUs (the 2 blue LEDs blinks) and the M4 green LED indicate that the Crazyflie is receiving radio packets,.

The problem then most likely comes from the software running on your PC. You can try to make sure you are running the latest version of the client. This can be done by running in the console:
```
pip install --upgrade cfclient
python -m cfclient.gui
```

Look in the console when running the bootloader to see if there is any error printed there.

**Reply by Paramjit26** [reply]
Thank you sir, 

I was able to solve the issue with one crazyflie. But with other Crazyflie its not even connecting with cfClient

https://github.com/bitcraze/discussions/assets/137099045/b56a842e-9b8c-42a9-b464-8a5f2bd0924f

**Comment by krichardsson** [comment]
Hi,
Great that one works!

Let's try to fix the other one as well. 
There are instructions for [recovery firmware flashing in the client docutmentation](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/). Have you tried this yet?

**Comment by dennybill** [comment]
> Hi, Great that one works!
> 
> Let's try to fix the other one as well. There are instructions for [recovery firmware flashing in the client](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/) [docutmentation](https://gbwa.download/). Have you tried this yet?

I recommend carefully reviewing the documentation and following the steps outlined for firmware recovery.
